### ISSUE TITLE
BUS/BUSI Domain Generalization Reproducibility

### ISSUE THREAD

**Issue by TalhaAhmed2000** [King-HAW/GMS#10]
Hello.

I have a query regarding the actual settings of the domain generalization experiment for BUS - BUSI & BUSI - BUS datasets. Currently I am doing the following setup:

BUSI to BUS 
- Load the pretrained SD-VAE weights provided (`768-v-ema-f...ge-VAE.ckpt`) and the pretrained LMM weights provided for BUSI (under the directory `ckpt/valid_busi`) then run inference on the test set of BUS (according to the pickle file its 31 samples)

**Results**: DSC: 72.95 but the paper shows DSC: 80.31.

BUS to BUSI
- Load the pretrained SD-VAE weights provided (`768-v-ema-f...ge-VAE.ckpt`) and the pretrained LMM weights provided for BUS (under the directory `ckpt/valid_bus`) then run inference on the test set of BUSI (according to the pickle file its 130 samples)

**Results**: DSC: 58.73 but the paper shows 61.60. 

As you can see, there is some discrepancy between my results and the papers. Could you clarify if the steps I have carried out is indeed what you did for the domain generalization experiment? If yes, could you maybe share some insight as to what might be missing in my code. If not, could you please share exactly how you carried out the domain generalization experiment. For e.g for the case BUS to BUSI:

Instead of using the weights in `ckpt/valid_bus`, did you train the LMM on the whole BUS dataset (i.e. train + test set = 132 + 31 samples) and test it on the test set of BUSI (130 samples) or the test it on the whole BUS dataset (train + test = 517 + 130 = 647 samples)? If yes, how did you perform validation during the LMM training and consequently model selection (i.e. `best_valid_dice.pth, best_valid_loss_all.pth` etc) (which is supposedly now different from the pth file in `ckpt/valid_busi`). OR did you train on the train set of BUS (132 samples) and test it on the whole BUSI dataset (647 samples)? Likewise please address the other way around i.e. BUSI to BUS. 

I would appreciate if you could be as specific as possible so that we can accurately reproduce the domain generalization results.

Thank you for your time and consideration.
